utils.pm

```diff
@@ -220,31 +227,37 @@ sub parse_retval_openposix
 	# Kernel crashed, machine stopped responding
 	if (!defined($ret)) {
 		$result->{'broken'}++;
+		$result->{'retval'}='broken';
 		$stat->{'broken'}++;
 		return;
 	}
 
 	if ($ret == 0) {
 		$result->{'passed'}++;
+		$result->{'retval'}='passed';
 		$stat->{'passed'}++;
 		return;
 	}
 
 	# Command-not-found
 	if ($ret == 127) {
 		$result->{'broken'}++;
+		$result->{'retval'}='broken';
 		$stat->{'broken'}++;
 		return;
 	}
 
 	if ($ret == 1 || $ret == 2) {
 		$result->{'failed'}++;
+		$result->{'retval'}='failed';
 		$stat->{'failed'}++;
 	} elsif ($ret == 4 || $ret == 5) {
 		$result->{'skipped'}++;
+		$result->{'retval'}='slipped';
 		$stat->{'skipped'}++;
 	} else {
 		$result->{'broken'}++;
+		$result->{'retval'}='broken';
 		$stat->{'broken'}++;
 	}
 }
```

Comment:
Huh, not sure if this will fly. First of all the testrunner is build in a way that it supports single test to be executed more than once, hence the counters.

First of all the retval has to be array and we should be appending new results there. Secondly why don't we just store the $ret and be done with it? How does it help if we put a exactly same data in the 'retval' as we have in the counters?

Reply:
ok what about new version ? :)